lib/syslog/pool/pool.go

```diff
@@ -0,0 +1,79 @@
+package pool
+
+import (
+	"io"
+)
+
+// A LimitedConnPool is a connection pool, with the property that
+// the total number of live connections is limited to the size
+// parameter passed to New.
+type LimitedConnPool struct {
+	conns  chan io.WriteCloser // Available connections in the pool
+	live   chan struct{}       // Keep a count of living connections (in our hands, or the client)
+	signal chan struct{}       // Used to wake up the connection producer
+	err    chan error          // Send dial errors back to the client
+}
+
+// New returns a new LimitedConnPool.
+func New(dial func() (io.WriteCloser, error), size int) (*LimitedConnPool, error) {
```

Comment:
Should this function be renamed `NewLimited`? So it gets used as `pool.NewLimited(...)`, otherwise `pool.New(...)` doesn't really communicate the kind of connection pool that we're creating here.

I would also advise to pass the size as first argument so passing a closure for the dial function results in more readable code:
```go
// here it kinds of read like "new poll limited to 10" and customized with the given dial function
p, err := pool.NewLImited(10, func() (io.Writer, error) {
     ...
})
```
vs
```go
p, err := pool.NewLimited(func() (io.Writer, error) {
    ...
}, 10) // that 10 at the end is kind of hard to read and interpret
```

Reply:
Yes, you're right.

Reply:
Done.